src/cose/mac0.rs

```diff
@@ -0,0 +1,546 @@
+/// let signature = tag(signature_payload, &signer).unwrap();
+/// let cose_mac0 = prepared.finalize(signature);
+/// fn tag(signature_payload: &[u8], s: &Hmac<Sha256>) -> anyhow::Result<Vec<u8>> {
+///     let mut mac = s.clone();
+///     mac.reset();
+///     mac.update(signature_payload);
+///     Ok(mac.finalize().into_bytes().to_vec())
+///  }
+#[derive(Debug, Clone, FieldsNames)]
+pub struct PreparedCoseMac0 {
+    tagged: bool,
+    cose_mac0: CoseMac0,
+    tag_payload: Vec<u8>,
+}
+
+impl CborSerializable for PreparedCoseMac0 {}
+impl AsCborValue for PreparedCoseMac0 {
+    fn from_cbor_value(value: Value) -> coset::Result<Self> {
+        let mut map = value
+            .into_map()
+            .map_err(|_| {
+                CoseError::DecodeFailed(ciborium::de::Error::Semantic(
+                    None,
+                    "PreparedCoseSign1 is not a map".to_string(),
+                ))
+            })?
+            .into_iter()
+            .map(|(k, v)| (k.into(), v.into()))
+            .collect::<BTreeMap<CborValue, CborValue>>();
+        Ok(PreparedCoseMac0 {
+            tagged: map
+                .remove(&CborValue::Text(PreparedCoseMac0::fn_tagged().to_string()))
+                .ok_or(CoseError::DecodeFailed(ciborium::de::Error::Semantic(
+                    None,
+                    "tagged not found".to_string(),
+                )))?
+                .into_bool()
+                .map_err(|_| {
+                    CoseError::DecodeFailed(ciborium::de::Error::Semantic(
+                        None,
+                        "tagged is not a bool".to_string(),
+                    ))
+                })?,
+            cose_mac0: CoseMac0::from_cbor_value(
+                map.remove(&CborValue::Text(
+                    PreparedCoseMac0::fn_cose_mac0().to_string(),
+                ))
+                .ok_or(CoseError::DecodeFailed(ciborium::de::Error::Semantic(
+                    None,
+                    "cose_sign1 not found".to_string(),
+                )))?
+                .into(),
+            )?,
+            tag_payload: map
+                .remove(&CborValue::Text(
+                    PreparedCoseMac0::fn_tag_payload().to_string(),
+                ))
+                .ok_or(CoseError::DecodeFailed(ciborium::de::Error::Semantic(
+                    None,
+                    "signature_payload not found".to_string(),
+                )))?
+                .into_bytes()
+                .map_err(|_| {
+                    CoseError::DecodeFailed(ciborium::de::Error::Semantic(
+                        None,
+                        "signature_payload is not bytes".to_string(),
+                    ))
+                })?,
+        })
+    }
+
+    fn to_cbor_value(self) -> coset::Result<Value> {
+        Ok(Value::Map(vec![
+            (
+                Value::Text(PreparedCoseMac0::fn_tagged().to_string()),
+                Value::Bool(self.tagged),
```

Comment:
Why is this serialized? If fact, do we need to serialize `PreparedCoseMac0` at all?

Reply:
There was serialized before, normally we don't need it as this is intermediary object from which we build `Cose*` ones. But as it was before I wanted to keep it.